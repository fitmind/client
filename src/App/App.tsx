import React from 'react';
import './App.css';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import store, { history } from '../redux/store';
import AppRouter from './AppRouter/AppRouter';
import { ConnectedRouter } from 'connected-react-router';

const theme = {
    main: 'mediumseagreen',
};

const App: React.FC = (): JSX.Element => {
    return (
        <Provider store={store}>
            <ConnectedRouter history={history}>
                <ThemeProvider theme={theme}>
                    <AppRouter />
                </ThemeProvider>
            </ConnectedRouter>
        </Provider>
    );
};

export default App;
