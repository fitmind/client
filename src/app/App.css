@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,300i,400,400i,600,600i,700,700i,800,800i|Yeseva+One');

html {
    font-family: 'Open Sans', Fallback, sans-serif;
    font-size: 16px;
    line-height: 1.2;
}

h1,
h2,
h3 {
    font-family: 'Yeseva One', Fallback, sans-serif;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace;
}
