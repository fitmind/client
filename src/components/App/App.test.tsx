import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('App', () => {
    // simple smoke tests - this will pass if the component renders successfully, even when an inner component throws an error
    it('renders without crashing', () => {
        shallow(<App />);
    });
    // example showing how to test that other components live inside a component
    it('should match the snapshot', () => {
        const component = renderer.create(<App />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
