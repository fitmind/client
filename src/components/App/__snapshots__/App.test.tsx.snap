// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <span
      className="navbar-brand"
      onClick={[Function]}
    >
      FITMIND ADMIN
    </span>
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Login
      </a>
      <a
        className="nav-link"
        data-rb-event-key={null}
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Dashboard
      </a>
    </div>
  </nav>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-md-3"
      />
      <div
        className="col"
      >
        <div
          className="sc-htpNat fYCKZt"
        >
          <div
            className="card border-secondary"
          >
            <h5
              className="card-header"
            >
              Admin Login
            </h5>
            <div
              className="card-body"
            >
              <p
                className="card-text"
              >
                Please enter with your admin login credentials
              </p>
              <form
                className=""
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="formBasicEmail"
                  >
                    Email address
                  </label>
                  <input
                    className="form-control"
                    id="formBasicEmail"
                    placeholder="Enter email"
                    type="email"
                  />
                  <small
                    className="text-muted form-text"
                  >
                    You can only login as an admin.
                  </small>
                </div>
                <div
                  className="form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="formBasicPassword"
                  >
                    Password
                  </label>
                  <input
                    className="form-control"
                    id="formBasicPassword"
                    placeholder="Password"
                    type="password"
                  />
                </div>
                <button
                  className="btn btn-outline-primary"
                  disabled={false}
                  type="submit"
                >
                  Login
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-md-3"
      />
    </div>
  </div>
</div>
`;
